### ISSUE TITLE
Inconsistent list of files being tracked between git lfs status and git lfs ls-files

### ISSUE THREAD

**Issue by NuLL3rr0r** [git-lfs/git-lfs#4581]
**Describe the bug**
Well, I have a huge repository (in GBs) described in #4559, which over time I noticed some files are not being tracked by LFS due to the fact that <code>git count-objects -vH</code> reported the drastic size growth with each commit. So, I decided to create a new repository from scratch and run some tests. Here are the results with an example:

1. I first initialized the repo and added and committed my non-LFS changes by following [what I've written here](https://www.babaei.net/blog/host-unreal-engine-projects-microsoft-azure-devops-unlimited-cost-free-git-lfs-quota/).

2. I ran the script mentioned in the article and it added <code>84,218</code> LFS objects and tracked them through the <code>.gitattributes</code> file. e.g:
```
"Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat" filter=lfs diff=lfs merge=lfs -text
```

3. Since the amount of files being tracked, I tried to diff the changes and found some files, which appear on <code>git lfs status</code> are not appeared on <code>git lfs ls-files</code> output:

```bash
$ git lfs status | wc -l # TO GET A ROUGH NUMBER OF FILES TRACKED BY LFS
84229

$ git lfs status | grep "icudtl\.dat"
	Engine/Binaries/ThirdParty/CEF3/Linux/Resources/icudtl.dat (LFS: e9933d8)
	Engine/Binaries/ThirdParty/CEF3/Linux/icudtl.dat (LFS: e9933d8)
	Engine/Binaries/ThirdParty/CEF3/Mac/Chromium Embedded Framework.framework/Resources/icudtl.dat (LFS: e9933d8)
	Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat(LFS: e9933d8) # THIS IS THE FILE BEING TRACKED BY .gitattributes
	Engine/Plugins/Bridge/ThirdParty/Mac/node-bifrost.app/Contents/Frameworks/Electron Framework.framework/Resources/icudtl.dat (LFS: 8364e6c)
	Engine/Source/ThirdParty/CEF3/cef_binary_3.2623.1395.g3034273_linux64/Resources/icudtl.dat (LFS: 2857fbe)
	Engine/Source/ThirdParty/CEF3/cef_binary_84.1.6+gc551bc2+chromium-84.0.4147.38_windows32/Resources/icudtl.dat (LFS: e9933d8)
	Engine/Source/ThirdParty/CEF3/cef_binary_84.1.6+gc551bc2+chromium-84.0.4147.38_windows64/Resources/icudtl.dat (LFS: e9933d8)
	Engine/Source/ThirdParty/CEF3/cef_binary_84.4.0+g304e015+chromium-84.0.4147.105_macosx64/Debug/Chromium Embedded Framework.framework/Resources/icudtl.dat (LFS: e9933d8)
	Engine/Source/ThirdParty/CEF3/cef_binary_84.4.0+g304e015+chromium-84.0.4147.105_macosx64/Release/Chromium Embedded Framework.framework/Resources/icudtl.dat (LFS: e9933d8)
	Engine/Source/ThirdParty/CEF3/cef_binary_84.4.1+gfdc7504+chromium-84.0.4147.105_linux64/Resources/icudtl.dat (LFS: e9933d8)
	Engine/Source/ThirdParty/CEF3/cef_binary_84.4.1+gfdc7504+chromium-84.0.4147.105_windows32/Resources/icudtl.dat (LFS: e9933d8)
	Engine/Source/ThirdParty/CEF3/cef_binary_84.4.1+gfdc7504+chromium-84.0.4147.105_windows64/Resources/icudtl.dat (LFS: e9933d8)

$ git lfs ls-files | wc -l # WHY ARE THESE NUMBERS INCONSISTENT?            
60851

$ git lfs ls-files | grep "icudtl\.dat" # WHERE THE HELL DID Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat AND THE REST OF icudtl.dat files FROM ABOVE GO?
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Linux/Resources/icudtl.dat
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Linux/icudtl.dat
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Mac/Chromium Embedded Framework.framework/Resources/icudtl.dat
8364e6c6bf * Engine/Plugins/Bridge/ThirdParty/Mac/node-bifrost.app/Contents/Frameworks/Electron Framework.framework/Resources/icudtl.dat
2857fbe46d * Engine/Source/ThirdParty/CEF3/cef_binary_3.2623.1395.g3034273_linux64/Resources/icudtl.dat
```

Well, I guess it's obvious <code>git lfs status</code> claims some files are being tracked by you cannot find them in the <code>git lfs ls-files</code> output. For the most part it tracks the files in <code>.gitattributes</code> and ignores around <code>~24,000</code> of them.

**To Reproduce**

Well, it's overwhelming since you have to follow [this blog post](https://www.babaei.net/blog/host-unreal-engine-projects-microsoft-azure-devops-unlimited-cost-free-git-lfs-quota/). But, I can temporarily create a public repository and you could run the script which for the first time may take hours to reproduce.

**Expected behavior**

Git LFS should track <code>84,218</code> LFS objects added through <code>.gitattributes</code>. That file is huge so, I'll attach a compressed version. Maybe you could use the <code>touch</code> command with an script to reproduce those files in a pseudo repository. 
[gitattributes.tar.gz](https://github.com/git-lfs/git-lfs/files/6988024/gitattributes.tar.gz)

**System environment**
The version of your operating system, plus any relevant information about platform or configuration (e.g., container or CI usage, Cygwin, WSL, or non-Basic authentication).  If relevant, include the output of `git config -l` as a code block.

```bash
$ git config -l
color.branch=auto
color.diff=auto
color.grep=auto
color.interactive=auto
color.showbranch=auto
color.status=auto
commit.gpgsign=true
core.editor=nvim
github.user=NuLL3rr0r
init.defaultbranch=master
pull.ff=only
pull.rebase=true
user.email=OMITTED
user.name=Mamadou Babaei
user.signingkey=OMITTED
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f

$ uname -a
Linux mamadou-pc 5.10.58 #55 SMP Sat Aug 14 18:42:12 CEST 2021 x86_64 Intel(R) Core(TM) i7-6820HK CPU @ 2.70GHz GenuineIntel GNU/Linux
```

**Output of `git lfs env`**

```bash
$ git lfs env
git-lfs/2.13.3 (GitHub; linux amd64; go 1.16.3)
git version 2.32.0

Error reading git config: error running /usr/libexec/git-core/git 'rev-parse' '--is-bare-repository': 'fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).' 'exit status 128'
LocalWorkingDir=
LocalGitDir=
LocalGitStorageDir=
LocalMediaDir=lfs/objects
LocalReferenceDirs=
TempDir=lfs/tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file
UploadTransfers=basic,lfs-standalone-file
GIT_EXEC_PATH=/usr/libexec/git-core
git config filter.lfs.process = ""
git config filter.lfs.smudge = ""
git config filter.lfs.clean = ""
```

**Additional context**

**Comment by NuLL3rr0r** [comment]
I must also mention that running the following it says that file is being tracked:

```bash
$ git lfs track --filename Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat
"Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat" already supported
```

I also tried to modify the <code>.gitattributes</code> file and remove the following line:

```
"Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat" filter=lfs diff=lfs merge=lfs -text
```

Then run the following:

```bash
$ git lfs track --filename Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat
Tracking "Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat"

$ git diff .gitattributes
diff --git a/.gitattributes b/.gitattributes
index 87e034340..4e9043baf 100644
--- a/.gitattributes
+++ b/.gitattributes
@@ -1,15 +1,11 @@
 #
 # This file contains rules that control how Git handles binary and text files, including line endings
 #
-
-
 # Make sure Windows batch files preserve CR/LF line endings, otherwise they may not be able to execute.  Windows
 # batch files require a CR/LF for labels to work properly, otherwise they may fail when labels straddle 512-byte
 # block boundaries.  This is important when files are downloaded through a zip archive that was authored on a
 # Linux machine (the default behavior on GitHub)
-
 *.bat eol=crlf
-
 # UE4 Git Dependencies
 ".tgitconfig" filter=lfs diff=lfs merge=lfs -text
 ".ue4dependencies" filter=lfs diff=lfs merge=lfs -text
@@ -334,7 +330,6 @@
 "Engine/Binaries/ThirdParty/CEF3/Win64/chrome_elf.dll" filter=lfs diff=lfs merge=lfs -text
 "Engine/Binaries/ThirdParty/CEF3/Win64/d3dcompiler_43.dll" filter=lfs diff=lfs merge=lfs -text
 "Engine/Binaries/ThirdParty/CEF3/Win64/d3dcompiler_47.dll" filter=lfs diff=lfs merge=lfs -text
-"Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat" filter=lfs diff=lfs merge=lfs -text
 "Engine/Binaries/ThirdParty/CEF3/Win64/libEGL.dll" filter=lfs diff=lfs merge=lfs -text
 "Engine/Binaries/ThirdParty/CEF3/Win64/libGLESv2.dll" filter=lfs diff=lfs merge=lfs -text
 "Engine/Binaries/ThirdParty/CEF3/Win64/libcef.dll" filter=lfs diff=lfs merge=lfs -text
@@ -84229,3 +84224,4 @@
 "Templates/TemplateResources/Standard/VehicleAdv/Media/VehicleAdv.png" filter=lfs diff=lfs merge=lfs -text
 "Templates/TemplateResources/Standard/VehicleAdv/Media/VehicleAdv_Preview.png" filter=lfs diff=lfs merge=lfs -text
 "cpp.hint" filter=lfs diff=lfs merge=lfs -text
+Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat filter=lfs diff=lfs merge=lfs -text
```

This clearly proves something is wrong:

``` bash
$ git reset Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat
$ git add -f Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat
$ git lfs ls-files | grep "icudtl\.dat" # STILL NO NEWS ABOUT Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Linux/Resources/icudtl.dat
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Linux/icudtl.dat
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Mac/Chromium Embedded Framework.framework/Resources/icudtl.dat
8364e6c6bf * Engine/Plugins/Bridge/ThirdParty/Mac/node-bifrost.app/Contents/Frameworks/Electron Framework.framework/Resources/icudtl.dat
2857fbe46d * Engine/Source/ThirdParty/CEF3/cef_binary_3.2623.1395.g3034273_linux64/Resources/icudtl.dat
```

**Comment by NuLL3rr0r** [comment]
Well, this is another update. I tried to commit the changes to see if it makes any difference. Guess what? After the commit:

```bash
git lfs ls-files | grep "icudtl\.dat"                                    
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Linux/Resources/icudtl.dat
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Linux/icudtl.dat
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Mac/Chromium Embedded Framework.framework/Resources/icudtl.dat
e9933d8b99 * Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat
8364e6c6bf * Engine/Plugins/Bridge/ThirdParty/Mac/node-bifrost.app/Contents/Frameworks/Electron Framework.framework/Resources/icudtl.dat
2857fbe46d * Engine/Source/ThirdParty/CEF3/cef_binary_3.2623.1395.g3034273_linux64/Resources/icudtl.dat
e9933d8b99 * Engine/Source/ThirdParty/CEF3/cef_binary_84.1.6+gc551bc2+chromium-84.0.4147.38_windows32/Resources/icudtl.dat
e9933d8b99 * Engine/Source/ThirdParty/CEF3/cef_binary_84.1.6+gc551bc2+chromium-84.0.4147.38_windows64/Resources/icudtl.dat
e9933d8b99 * Engine/Source/ThirdParty/CEF3/cef_binary_84.4.0+g304e015+chromium-84.0.4147.105_macosx64/Debug/Chromium Embedded Framework.framework/Resources/icudtl.dat
e9933d8b99 * Engine/Source/ThirdParty/CEF3/cef_binary_84.4.0+g304e015+chromium-84.0.4147.105_macosx64/Release/Chromium Embedded Framework.framework/Resources/icudtl.dat
e9933d8b99 * Engine/Source/ThirdParty/CEF3/cef_binary_84.4.1+gfdc7504+chromium-84.0.4147.105_linux64/Resources/icudtl.dat
e9933d8b99 * Engine/Source/ThirdParty/CEF3/cef_binary_84.4.1+gfdc7504+chromium-84.0.4147.105_windows32/Resources/icudtl.dat
e9933d8b99 * Engine/Source/ThirdParty/CEF3/cef_binary_84.4.1+gfdc7504+chromium-84.0.4147.105_windows64/Resources/icudtl.dat

$ git count-objects -vH                           
count: 0
size: 0 bytes
in-pack: 201402
packs: 1
size-pack: 308.71 MiB
prune-packable: 0
garbage: 0
size-garbage: 0 bytes

$ git lfs ls-files | wc -l                                                 
84135
```

Before the commit, <code>size</code> was around 460/470 MBs, and size-pack: was between 296/297 MBs if I recall correctly. The interesting part is after the commit, <code>git lfs ls-files</code> now reports the correct amount of tracked files and actually I can see <code>Engine/Binaries/ThirdParty/CEF3/Win64/icudtl.dat</code> in the output.

I'll try to experiment more as I update this repo in the following days and post my findings. But, I guess this is still a bug considering the different outputs before and after the commit.

**Comment by bk2204** [comment]
Hey,

It does sound like there's something fishy going on here, but I'm not sure what it is.  Unfortunately, as we've discussed in the past, we need a simple way to reproduce problems, ideally with a single shell script without any dependencies, and we don't currently have that.  Your shell script has several dependencies, including the Unreal Engine, which we don't have access to.

Could we ask you to try to provide a minimal, complete, verifiable example that demonstrates the problem?

**Comment by NuLL3rr0r** [comment]
I do understand the Unreal Engine and file size barrier. So, I'll try to find some time in the following days and reproduce it using a script that generates the same directory and file structure and arrangments. I will let you know of the results.

**Comment by NuLL3rr0r** [comment]
Hello again!

I've just wanted to give an update on this. It seems the issue was not Git LFS. Originally my script had some issues, which caused some files to get tracked by Git and not LFS. Even though I fixed the script the repo was messed up. So, with each update to those files, the size was getting accumulated.

Shortly after I opened this bug report, I created an Azure repository from scratch. Since then I have been updating the Engine regularly using my script and the size stayed reasonable:

```
$ git count-objects -vH
count: 0
size: 0 bytes
in-pack: 272188
packs: 1
size-pack: 341.33 MiB
prune-packable: 0
garbage: 0
size-garbage: 0 bytes
```

Sorry for the inconvenience and thank you for this awesome piece of software! Keep up the good work!
